internal/bgp/frr/frr.go

```diff
@@ -288,22 +299,26 @@ func (sm *sessionManager) createConfig() (*frrConfig, error) {
 			}
 
 			neighbor.Advertisements = append(neighbor.Advertisements, &advConfig)
-			if _, ok := routerAnnouncements[prefix]; !ok {
-				switch family {
-				case ipfamily.IPv4:
-					router.IPV4Prefixes = append(router.IPV4Prefixes, prefix)
-				case ipfamily.IPv6:
-					router.IPV6Prefixes = append(router.IPV6Prefixes, prefix)
-				}
-				routerAnnouncements[prefix] = struct{}{}
+			switch family {
+			case ipfamily.IPv4:
+				rout.ipV4Prefixes[prefix] = prefix
+			case ipfamily.IPv6:
+				rout.ipV6Prefixes[prefix] = prefix
```

Comment:
Can't we make `rout.ipV4Prefixes` and `rout.ipV6Prefixes` to be slices?
I don't think maps are required here.

Reply:
On top of that, I think that we should refactor the whole createConfig() function in the sense
of separating into smaller functions.

Reply:
The maps are to avoid duplicates

Reply:
> On top of that, I think that we should refactor the whole createConfig() function in the sense of separating into smaller functions.

I am not sure it would be better. I kind of like the fact the rendering mechanism is clear. The lenght is mostly because of literals that take multiple lines.

Reply:
Extracting to separate, smaller, and more atomic functions can only improve clarity IMO.
But anyways, I see you addressed all the comments I wrote above and it's LGTM from me. :)

Reply:
I am generally in favour of hiding complexity behind funcitons with meaningful names, but long function = not clear is not always necessarily true. If you need to jump between the loop and the subfunctions to understand the behaviour, the clarity is less.

This function is easy to follow, it's long but as I wrote, the biggest reason for the length is the literal initializiatoins that consume multiple rows.